tests/x11/steam.pm

```diff
@@ -23,8 +24,11 @@ sub run {
     # start steam in terminal to watch the progress on initial startup, e.g.
     # the update progress
     script_run 'steam', 0;
-    assert_and_click 'steam-accept-agreement';
-    assert_and_click 'steam-agreement-ok';
+    if (check_screen 'steam-accept-agreement', 30) {
```

Comment:
As per your verification run not to waste 30 seconds when license is not there, we could simply do:
```perl
assert_screen([qw(steam-accept-agreement steam-login steam-startup-libgl_error)], 600);
if(match_has_tag('steam-accept-agreement') {
    assert_and_click 'steam-accept-agreement';
    assert_and_click 'steam-agreement-ok';
    assert_screen([qw(steam-login steam-startup-libgl_error)], 600);
}
```
Also we could extract common needles from both assert_screen calls.

Reply:
done